### PR TITLE
refactor: pull signature and ttl validation out into separate file with it's own tests

### PR DESCRIPTION
**Pull request by ezavada**
Closes: WORLD-1237

## Overview

Cleaned up the validation code and put it into a separately testable validator.go file. Also added a number of tests that don't depend on the overall server infrastructure to operate.

## Release Notes

- **New Features**
  - Introduced a `SignatureValidator` for enhanced transaction validation, improving security and modularity.
  - Added a method to retrieve signer addresses based on persona tags.

- **Improvements**
  - Streamlined server configuration options for signature verification and message expiration.
  - Enhanced error handling and validation responses for transaction processing.
  - Added utility functions for handling timestamps in the signing package.

- **Tests**
  - Expanded test coverage for transaction signing, timestamp accuracy, and JSON validation.

### CODE REVIEW

File: cardinal/server/validator/validator.go [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+	validator := SignatureValidator{
+		IsDisabled:               disabled,
+		MessageExpirationSeconds: msgExpirationSec,
+		HashCacheSizeKB:          hashCacheSizeKB,
+		namespace:                namespace,
+		cache:                    nil,
+		signerAddressProvider:    provider,
+	}
+	if !disabled {
+		validator.cache = freecache.NewCache(validator.HashCacheSizeKB)
```
[the remark this answers is not in the document]
Comment (ezavada) [reply cardinal/server/validator/validator.go:80]: excellent catch, Code Rabbit!

File: cardinal/server/validator/validator.go [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+	}
+	// return all other errors
+	return false, err
+}
+
+// validateSignature validates that the signature of transaction is valid
+func (validator *SignatureValidator) validateSignature(tx *Transaction, signerAddr string) error {
+	if tx.Namespace != validator.namespace {
+		return eris.Wrap(ErrWrongNamespace, fmt.Sprintf("expected %q got %q", validator.namespace, tx.Namespace))
+	}
```
[the remark this answers is not in the document]
Comment (ezavada) [reply cardinal/server/validator/validator.go:176]: It's a reasonable suggestion, but I want to keep this function restricted to validating the TTL of the message.

**Comment by ezavada** [comment]
@smsunarto @rmrt1n git is unfortunately show this PR as if it included all my changes from the WORLD-1190 PR. Scroll way down to  [ec1973a](https://github.com/Argus-Labs/world-engine/pull/804/commits/ec1973ab42c9d2510edddd981a8d74750a3f47e8) to see the actual changes for this PR. Sorry about that, I'll wait to merge main until the reviews are done in the future.

**Comment by ezavada** [comment]
Alternatively, you can look at server/handler/tx.go, server/validator/ folder, world_persona.go, and sign/sign.go

All the changes were in those files (sign.go are comments and one new function for completeness of timestamp handling)

**Comment by smsunarto** [comment]
Just rebase the PR

**Pull request closed without merging**